### ISSUE TITLE
Clean up demand lib

### ISSUE THREAD

**Issue by simnh** [oemof/oemof#105]
* Docstrings are missing. (espacially in heating part)
* pep8 style 
* Adpat to oemof concept (remove database conn, classnames in CamelCase etc.)

**Comment by simonhilpert** [comment]
I created a new feature branch : features/cleanup_demandlib

Actually I think the energy_buildings.py could get some re-design. Starting with a building class and
deriving electrical / thermal demand would make sense to me (What is a 'ElectricalBuilding'?)

I think we should try to break the demand lib down to general functions and add data espacially specific data group in examples/ apps e.g.

Creating electrical/thermal demans based on a generic methodolgy with arbritray data -> oemof framework. 
Inserting e.g. specific data / connections to data from germany that we can not transfer to other regions -> app 

I feel that we need to work on our common understanding of the framework (I will open an extra issue)

@oemof/feedinlib  what do you think?

**Comment by cswh** [comment]
Regarding the general structure for a lib like demandlib and feedinlib and how it corresponds to the oemof entities, we should develop that with a high priority at the refactoring in April!

**Comment by uvchik** [comment]
So I change the milestone to the meeting milestone, okay?

**Comment by uvchik** [comment]
Could the "clean up" also cover the docstrings? Compiling the sphinx documentation locally, there are a lot of errors and warnings produced by the demandlib. I use the following command to create the documentation:
```bash
sphinx-build -b html .../git_home/oemof.db/doc .../git_home/oemof.db/doc/_build/html
```
This is the essential part of the sphinx output:
```python
reading sources... [100%] api/oemof.demandlib                                                                                                                                                                                                                                  
oemof/demandlib/demand.py:docstring of oemof.demandlib.demand.electrical_demand:29: ERROR: Unexpected indentation.
oemof/demandlib/demand.py:docstring of oemof.demandlib.demand.electrical_demand:30: WARNING: Block quote ends without a blank line; unexpected unindent.
oemof/demandlib/demand.py:docstring of oemof.demandlib.demand.electrical_demand:38: ERROR: Unexpected indentation.
oemof/demandlib/demand.py:docstring of oemof.demandlib.demand.electrical_demand:44: ERROR: Unexpected indentation.
oemof/demandlib/demand.py:docstring of oemof.demandlib.demand.electrical_demand:46: ERROR: Unexpected indentation.
oemof/demandlib/demand.py:docstring of oemof.demandlib.demand.electrical_demand:51: WARNING: Block quote ends without a blank line; unexpected unindent.
oemof/demandlib/demand.py:docstring of oemof.demandlib.demand.electrical_demand:53: ERROR: Unexpected indentation.
oemof/demandlib/demand.py:docstring of oemof.demandlib.demand.electrical_demand:62: ERROR: Unexpected indentation.
oemof/demandlib/demand.py:docstring of oemof.demandlib.demand.electrical_demand:70: ERROR: Unexpected indentation.
oemof/demandlib/demand.py:docstring of oemof.demandlib.demand.electrical_demand:85: WARNING: Field list ends without a blank line; unexpected unindent.
oemof/demandlib/demand.py:docstring of oemof.demandlib.demand.electrical_demand:89: WARNING: Inline strong start-string without end-string.
```

**Comment by gplssm** [comment]
We should get the docstring at least working independent from their content.

I would really appreciate if we could have a plan when to stop cleaning up the demandlib. I did some changes in the `cleanup-demandlib` branch that are necessary for my current daily work. On the other hand I start doing relevant things in other branches. This leads to some difficulties and at least confusion for me (which branch has to be checked out if I want to execute app X, app Y, ....).

I would suggest to assign this issue to March release and only do things we can to 'til then, merge it back and maybe open another cleanup branch afterwards, if required. What do you think?

**Comment by cswh** [comment]
Would be ok for me as gplssm suggests.

**Comment by ckaldemeyer** [comment]
+1

**Comment by uvchik** [comment]
:+1: 

@gplssm : Could you make a list of what will be done until the March release and what will be done later?

**Comment by gplssm** [comment]
Yep, I set up a separate [wiki page](https://github.com/oemof/oemof/wiki/Cleanup-demandlib-organization) for this. Please add topics on the agenda you like to work on or that are necessary to be worked on. :wink: 

This wiki page will be deleted afterwards to not mess up the wiki.

**Comment by gplssm** [comment]
@simnh any suggestions/ opinions from your side?

**Comment by simnh** [comment]
I do not use the demandlib except of the bdew heat functionality. For me it would make sense to 
keep implementations such as the bdew stuff in simple functions. I would like to keep this in a lean way....but after the next meeting in Flensburg I guess we can decide more on structure. 

I dont have any functionalities I would like to add / be added atm.

**Comment by gplssm** [comment]
Thanks for your input! So, I suggest we do clean up things we can manage this week and afterwards we merge cleanup branch (and sub-branches) back to dev.
During next meeting you can decide how to proceed.

**Comment by gplssm** [comment]
The according feature branch features/cleanup-demandlib is merged by 187a468a413f7d7813677116268a061a6dd013b9. Quality of docstrings is still improvable. The according work is coordinated in #123.
